ui/components/src/TimeChart/TimeChart.tsx

```diff
@@ -0,0 +1,364 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { DatasetOption } from 'echarts/types/dist/shared';
+import { forwardRef, MouseEvent, useImperativeHandle, useMemo, useRef, useState } from 'react';
+import { Box } from '@mui/material';
+import { utcToZonedTime } from 'date-fns-tz';
+import { getCommonTimeScale, TimeScale, UnitOptions, TimeSeries } from '@perses-dev/core';
+import type {
+  EChartsCoreOption,
+  GridComponentOption,
+  LineSeriesOption,
+  YAXisComponentOption,
+  TooltipComponentOption,
+} from 'echarts';
+import { ECharts as EChartsInstance, use } from 'echarts/core';
+import { LineChart as EChartsLineChart } from 'echarts/charts';
+import {
+  GridComponent,
+  DatasetComponent,
+  DataZoomComponent,
+  MarkAreaComponent,
+  MarkLineComponent,
+  MarkPointComponent,
+  TitleComponent,
+  ToolboxComponent,
+  TooltipComponent,
+  LegendComponent,
+} from 'echarts/components';
+import { CanvasRenderer } from 'echarts/renderers';
+import { EChart, OnEventsType } from '../EChart';
+import { ChartHandleFocusOpts, ChartHandle, TimeChartSeriesMapping } from '../model/graph';
+import { useChartsTheme } from '../context/ChartsThemeProvider';
+import {
+  clearHighlightedSeries,
+  enableDataZoom,
+  getFormattedAxisLabel,
+  getYAxes,
+  restoreChart,
+  ZoomEventData,
+} from '../utils';
+import { CursorCoordinates, TimeChartTooltip, TooltipConfig } from '../TimeSeriesTooltip';
+import { useTimeZone } from '../context/TimeZoneProvider';
+
+use([
+  EChartsLineChart,
+  GridComponent,
+  DatasetComponent,
+  DataZoomComponent,
+  MarkAreaComponent,
+  MarkLineComponent,
+  MarkPointComponent,
+  TitleComponent,
+  ToolboxComponent,
+  TooltipComponent,
+  LegendComponent,
+  CanvasRenderer,
+]);
+
+export interface TimeChartProps {
+  height: number;
+  data: TimeSeries[];
+  seriesMapping: TimeChartSeriesMapping;
```

Comment:
What is a seriesMapping?

Reply:
As part of this PR, I decided to switch to use ECharts `dataset` so that raw time series data could be passed into the chart instead of having loops for graphData.xAxis and individual series storing their own values. So all data is stored inside dataset.source and the "styles" are separate and defined in `seriesMapping`. 

`seriesMapping` then uses datasetId instead of storing the data directly in the series object. I guess the downside here is embed users have to know a bit about how ECharts `dataset` works, but overall, I do like the approach and this tutorial explains the tradeoffs: https://apache.github.io/echarts-handbook/en/concepts/dataset/#define-data-in-dataset